tests/experimental/pydantic/test_conversion.py

```diff
@@ -645,12 +656,120 @@ class UserInput:
 
+    definition: TypeDefinition = UserInput._type_definition
+    assert definition.name == "UserInput"
+
+    [
+        age_field,
+        password_field,
+    ] = (
+        definition.fields
+    )  # fields without a default go first, so the order gets reverse
+
+    assert age_field.python_name == "age"
+    assert age_field.type is int
+
+    assert password_field.python_name == "password"
+    assert isinstance(password_field.type, StrawberryOptional)
+    assert password_field.type.of_type is str
+
+
+def test_sort_creation_fields():
+    has_default = DataclassCreationFields(
+        name="has_default",
+        type_annotation=str,
+        field=StrawberryField(
+            python_name="has_default",
+            graphql_name="has_default",
+            default="default_str",
+            default_factory=UNSET,
+            type_annotation=str,
+            description="description",
+        ),
+    )
+    has_default_factory = DataclassCreationFields(
+        name="has_default",
+        type_annotation=str,
+        field=StrawberryField(
+            python_name="has_default",
+            graphql_name="has_default",
+            default=UNSET,
+            default_factory=lambda: "default_factory_str",
+            type_annotation=str,
+            description="description",
+        ),
+    )
+    no_defaults = DataclassCreationFields(
+        name="has_default",
+        type_annotation=str,
+        field=StrawberryField(
+            python_name="has_default",
+            graphql_name="has_default",
+            default=UNSET,
+            default_factory=UNSET,
+            type_annotation=str,
+            description="description",
+        ),
+    )
+    fields = [has_default, has_default_factory, no_defaults]
+    assert sort_creation_fields(fields) == [
+        no_defaults,
+        has_default,
+        has_default_factory,
+    ], "should place items with defaults last"
+
+
+def test_convert_input_types_to_pydantic_default_and_default_factory():
+    # Pydantic should raise an error if the user specifies both default
+    # and default_factory. this checks for a regression on their side
+    with pytest.raises(
+        ValueError,
+        match=("cannot specify both default and default_factory"),
+    ):
+
+        class User(pydantic.BaseModel):
+            password: Optional[str] = Field(default=None, default_factory=lambda: None)
+
+    # If the user defines both through a hacky way, we'll still going to catch it
+    with pytest.raises(
+        BothDefaultAndDefaultFactoryDefinedError,
+        match=("Not allowed to specify both default and default_factory."),
+    ):
+        hacked_field = Field(default_factory=lambda: None)
+
+        class User2(pydantic.BaseModel):
+            password: Optional[str] = hacked_field
+
+        hacked_field.default = None
+
+        @strawberry.experimental.pydantic.input(User2)
+        class UserInput:
+            password: strawberry.auto
```

Comment:
`with` scope should be narrowed to the exact line that raises the exception

Reply:
thanks!